spec/osut_tests_spec.rb

```diff
@@ -1333,1134 +1129,2447 @@ module M
+    #   - remove mini floor @Z0 and mini ceiling @Z3.05
+    #   - attic mini floor facing outdoors
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[0]
+    vtx << mini_floor_vtx[3]
+    vtx << mini_floor_vtx[0]
+    vtx << mini_ceiling_vtx[3]
+    mini_north = OpenStudio::Model::Surface.new(vtx, model)
+    mini_north.setName("Mini north")
+    expect(mini_north.setSpace(core)).to be(true)
+    expect(mini_north.outsideBoundaryCondition).to eq("Outdoors")
+
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[3]
+    vtx << mini_floor_vtx[0]
+    vtx << mini_floor_vtx[1]
+    vtx << mini_ceiling_vtx[2]
+    mini_east = OpenStudio::Model::Surface.new(vtx, model)
+    mini_east.setName("Mini east")
+    expect(mini_east.setSpace(core)).to be(true)
+    expect(mini_east.outsideBoundaryCondition).to eq("Outdoors")
+
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[2]
+    vtx << mini_floor_vtx[1]
+    vtx << mini_floor_vtx[2]
+    vtx << mini_ceiling_vtx[1]
+    mini_south = OpenStudio::Model::Surface.new(vtx, model)
+    mini_south.setName("Mini south")
+    expect(mini_south.setSpace(core)).to be(true)
+    expect(mini_south.outsideBoundaryCondition).to eq("Outdoors")
+
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[1]
+    vtx << mini_floor_vtx[2]
+    vtx << mini_floor_vtx[3]
+    vtx << mini_ceiling_vtx[0]
+    mini_west = OpenStudio::Model::Surface.new(vtx, model)
+    mini_west.setName("Mini west")
+    expect(mini_west.setSpace(core)).to be(true)
+    expect(mini_west.outsideBoundaryCondition).to eq("Outdoors")
+
+    mini_floor.remove
+    mini_ceiling.remove
+    expect(mini_attic.setOutsideBoundaryCondition("Outdoors")).to be(true)
+
+    # Re-validating pre-tested areas + volumes.
+    expect(core.floorArea).to be_within(TOL).of(149.66 - 4) # -mini m2
+    core_volume = core.floorArea * core.ceilingHeight
+    expect(core_volume).to be_within(TOL).of(core.volume)
+    expect(attic.volume).to be_within(TOL).of(720.19)
+    expect(attic.floorArea).to be_within(TOL).of(567.98)
+
+    expect(mod1.poly(core_floor, true).empty?).to be(true)   # now concave
+    expect(mod1.poly(core_ceiling, true).empty?).to be(true) # now concave
+    expect(mod1.poly(attic_floor, true).empty?).to be(true)  # now concave
+    expect(mod1.logs.size).to eq(3)
+    expect(mod1.error?).to be(true)
+    mod1.logs.each {|l| expect(l[:message].include?("non-convex")).to be(true)}
+    expect(mod1.clean!).to eq(DBG)
+
+    file = File.join(__dir__, "files/osms/out/mini2.osm")
+    model.save(file, true)
+    # Simulation results E+ 22.2 (SutherlandHodgman, PolygonClipping)
+    # ... no E+ errors/warnings.
+    #
+    # mini2.osm A  FullInteriorAndExterior                 248 GJ 3.8 UMH cool
+    # mini2.osm B  FullExteriorWithReflections             249 GJ 3.7 UMH cool
+    # mini2.osm C  FullInteriorAndExteriorWithReflections  249 GJ 3.7 UMH cool
```

Comment:
Slight expected increase in energy use, yet consistent between options. No related E+ 22.2 errors/warnings.